cmd/gen-test-vectors/main.go

```diff
@@ -0,0 +1,236 @@
+			userID:         "bob",
+			signers:        signers1,
+			authorizedKeys: authorizedKeys1,
+		},
+		{
+			desc:           "set_carol",
+			wantProfile:    nil,
+			setProfile:     []byte("carol-key1"),
+			ctx:            authentication.WithOutgoingFakeAuth(ctx, "carol"),
+			userID:         "carol",
+			signers:        signers1,
+			authorizedKeys: authorizedKeys1,
+		},
+		{
+			desc:           "bob1_get",
+			wantProfile:    []byte("bob-key1"),
+			setProfile:     nil,
+			ctx:            context.Background(),
+			userID:         "bob",
+			signers:        signers1,
+			authorizedKeys: authorizedKeys1,
+		},
+		{
+			desc:           "bob1_set",
+			wantProfile:    []byte("bob-key1"),
+			setProfile:     []byte("bob-key2"),
+			ctx:            authentication.WithOutgoingFakeAuth(ctx, "bob"),
+			userID:         "bob",
+			signers:        signers1,
+			authorizedKeys: authorizedKeys1,
+		},
+	} {
+		// Check profile.
+		e, err := env.Cli.GetEntry(ctx, &pb.GetEntryRequest{
+			DomainId: env.Domain.DomainId,
+			UserId:   tc.userID,
+			AppId:    appID,
+		})
+		if err != nil {
+			return fmt.Errorf("GetEntry(): %v", err)
+		}
+		if _, _, err := env.Client.VerifyGetEntryResponse(ctx, env.Domain.DomainId, appID, tc.userID, types.LogRootV1{}, e); err != nil {
+			return fmt.Errorf("VerifyGetEntryResponse(): %v", err)
+		}
+		if got, want := e.GetCommitted().GetData(), tc.wantProfile; !bytes.Equal(got, want) {
+			return fmt.Errorf("VerifiedGetEntry(%v): %s, want %s", tc.userID, got, want)
+		}
+		getEntryResps = append(getEntryResps, testdata.GetEntryResponseVector{
+			Desc:   tc.desc,
+			AppID:  appID,
+			UserID: tc.userID,
+			Resp:   e,
+		})
+
+		// Update profile.
+		if tc.setProfile != nil {
+			u := &tpb.User{
+				DomainId:       env.Domain.DomainId,
+				AppId:          appID,
+				UserId:         tc.userID,
+				PublicKeyData:  tc.setProfile,
+				AuthorizedKeys: tc.authorizedKeys,
+			}
+			cctx, cancel := context.WithTimeout(tc.ctx, env.Timeout)
+			defer cancel()
+			m, err := env.Client.Update(cctx, u, tc.signers)
+			if got, want := err, context.DeadlineExceeded; got != want {
+				return fmt.Errorf("Update(%v): %v, want %v", tc.userID, got, want)
+			}
+			cctx, cancel = context.WithTimeout(tc.ctx, env.Timeout)
+			defer cancel()
+			env.Receiver.Flush(cctx)
+			cctx, cancel = context.WithTimeout(tc.ctx, env.Timeout)
+			defer cancel()
+			if _, err := env.Client.WaitForUserUpdate(cctx, m); err != nil {
+				return fmt.Errorf("WaitForUserUpdate(%v): %v, want nil", m, err)
+			}
+		}
+		if err := SaveTestVectors(env, getEntryResps); err != nil {
+			return fmt.Errorf("SaveTestVectors(): %v", err)
+		}
+	}
+	return nil
+}
+
+// SaveTestVectors generates test vectors for interoprability testing.
+func SaveTestVectors(env *integration.Env, resps []testdata.GetEntryResponseVector) error {
+	// Output all key material needed to verify the test vectors.
+	domainFile := *testdataDir + "/domain.json"
```

Comment:
Typically flag values should only be accessed and used in main(), and then passed down to other functions that may need them as a parameter.  So in this case you would add an extra param to GenerateTestVectors and SaveTestVectors, and pass the testdata dir down through that.

Reply:
Good catch